### PR TITLE
Use field specific wrappers

### PR DESCRIPTION
**Pull request by MichalRemis**
Implemented field's specific wrapper_names to be passed to and used by CSV JS FormBuilder for adding/removing of error messages. Before CSV used only form-wide wrapper but SimpleForm's field may use custom wrapper specified:

- by field's `wrapper` attribute
- by field's type and `wrapper_mappings` attribute
- by field's type and `SimpleForm.config.wrapper_mappings`

I think this feature will be also required for implementation of validating of radios/dates/etc.

I am passing wrapper_name by input's `data-simple-form-wrapper`attribute. Was thinking about putting it into CSV Hash, but I think it would require CSV code change and this is SimpleForm's thing anyway.

### CODE REVIEW

**Comment by tagliala** [comment]
I think we can ignore the codeclimate offences

File: lib/client_side_validations/simple_form/form_builder.rb [hunk 1]
```diff
@@ -34,6 +36,14 @@ def wrapper_error_component
           wrapper.find(:full_error)
         end
       end
+
+      def add_field_specific_wrapper_name_to_field_options(attribute_name, options, &block)
+        wrapper_name = options[:wrapper] || find_wrapper_mapping(find_input(attribute_name, options, &block).input_type)
+        return if wrapper_name.nil?
+
+        options[:input_html] ||= {}
+        options[:input_html][:'data-simple-form-wrapper'] = wrapper_name
```
Comment (tagliala) [inline lib/client_side_validations/simple_form/form_builder.rb:45]: Amazing, thanks.

TL;DR: `[:'data-client-side-validations-wrapper']`, move logic to the main gem, use `super` here 

I think that this should be moved to the main gem, because it could be useful even without Simple Form. Then the simple form plugin could only edit options `find_wrapper_mapping(find_input(attribute_name, options, &block).input_type)` and call `super` 

 I would like to keep using the not-so-short `data-client-side-validations` attribute name (IMHO `data-simple-form` gives the idea that SimpleForm added the data attribute, violating the principle of least surprise) and `wrapper` is ok

Feel free to keep doing your experiments here, then you can move to the main gem or I can do it

Comment (MichalRemis) [reply lib/client_side_validations/simple_form/form_builder.rb]: Great, thanks for feedback. I agree with `:'data-client-side-validations-wrapper'` and updated it. Feel free to move this to the main gem. I continue by making radios, checkboxes etc. to work on new branch.

**Comment by tagliala** [comment]
Please rebase on the current master, this PR is not clean

If you have troubles in rebase, let me know

Short-not-tested rebase guide

```
git checkout master
git add upstream https://github.com/DavyJonesLocker/client_side_validations-simple_form.git
git pull upstream master # now your master branch is up-to-date with upstream
git checkout UseFieldSpecificWrappers
git rebase -- master
```

Follow on screen instructions.

```
git push -f
```

After the rebase, I should not see anymore the following commits in this PR:

image

**Comment by MichalRemis** [comment]
ok, rebased

**Comment by tagliala** [comment]
@MichalRemis thanks! Now it looks good

File: lib/client_side_validations/simple_form/form_builder.rb [hunk 2]
```diff
@@ -22,6 +22,8 @@ def input(attribute_name, options = {}, &block)
           options.delete(:validate)
         end
 
+        add_field_specific_wrapper_name_to_field_options(attribute_name, options, &block)
```
Comment (tagliala) [inline lib/client_side_validations/simple_form/form_builder.rb:25]: Moving this to the parent gem will not be straightforward. I think I should add the method call to all the overrides

Comment (MichalRemis) [reply lib/client_side_validations/simple_form/form_builder.rb:25]: Yes, I agree
